### DISCUSSION TITLE
New builds Q4 2023

### DISCUSSION THREAD

**Discussion by ianw1974** [ianw1974/zimbra-build-scripts#50] [Announcements]
# New builds for Q4 2023

New builds have been made for Q4 2023, replacing the ones made at the beginning of October.

## Why the change?

The reason for the change, is two-fold.  The first, ```BUILD_NO``` is now being updated to reflect when the build is made.  Since I normally build once per quarter, the four digit number which up until now has always been ```0001``` is now being replaced.  The first two digits referring to the quarter.  The second two digits being the year.  Therefore for the these builds ```0423```.

The second change is relating to the version.  It was previously shown as ```10.0.0``` which wasn't too indicative of the version.  Sure we knew it was Zimbra 10, however we wouldn't know if it was a minor point release - with the current point release being ```10.0.5```.  This has now been altered during the build process to make the version easily identifiable.

## What if multiple builds are made during the same quarter?

This is easily addressed, since the build always has a date/timestamp in the filename, which will be clear in these builds starting with ```20231031```.  Therefore there is no real need to change much else at this point in time.

## Why are patch releases, eg: P42 not shown?

The reason for this when building from source is that we also haven't built at the exact date/time as Zimbra to actually guarantee that the build doesn't differ in any other way.  Also due to the fact Zimbra utilises deb or rpm repositories for delivering patch updates on Network Edition of which isn't available for OSE/FOSS users, we have to patch at source by addressing the fixes that need to be applied.  At this point, whilst technically similar to their patch release, it isn't completely.  So there is no plan to **claim** that the builds are the exact same build release as Zimbra's Network Edition when they will most likely vary in a number of ways.  There is no plan to divert from this because it simply doesn't make too much sense.

Building from tags has always been hit-and-miss as well, which is why the scripts provided here build by default from the develop branch.  The builds I provide are also built from this branch, and again, there is no plan to change this.  What is more important at this time is that the builds address the issues that need to be fixed.  There is no plan to introduce my own patch system either.

Probably the easiest thing is just to post and say in an announcement like this, what issues have been addressed in that particular build rather than attempt to claim matching Zimbra's patch number.

## Where can I get the builds from?

You can find them here: https://techfiles.online/zimbra

**Comment by jeredfloyd** [comment]
With the release of 10.0.6 (which resolves several CVEs), do you still plan to only do "release" builds quarterly?  (Thanks for the quick reply on #63 btw!)

**Reply by ianw1974** [reply]
The CVE's have been around since at least April 2022 if not longer in some cases.  They are marked as **high**.  I release quarterly builds unless there is a **critical** issue.  There's only 10 days to the end of the year when my Q1 2024 builds will be released.

I think waiting 10 days isn't bad, especially if we consider those CVE's are like 1.5 years old and only just addressed by Synacor/Zimbra.

**Reply by jeredfloyd** [reply]
Makes sense to me, for both policy and practicality.

The OpenJDK update is ancient, and I don't think has RCE or remote privilege escalation concerns.  I'm a little more curious about "Fixed a vulnerability where an auth token was possible to be obtained" with severity "TBD", but I hope that Zimbra would tell us if this was being actively exploited -- it looks like it's in the same class of previous privilege escalation attacks, but may be just cleanup.  Forums (https://forums.zimbra.org/viewtopic.php?p=312139#p312139) suggest this was removal of more obsolete files (in 8.8, at least) , which can be done manually on 10.0.5 if an issue exists:
```
/opt/zimbra/jetty_base/webapps/zimbra/public/access.jsp
/opt/zimbra/jetty_base/webapps/zimbra/public/authorize.jsp
/opt/zimbra/jetty_base/work/zimbra/jsp/org/apache/jsp/public_/access_jsp.class
/opt/zimbra/jetty_base/work/zimbra/jsp/org/apache/jsp/public_/access_jsp.java
/opt/zimbra/jetty_base/work/zimbra/jsp/org/apache/jsp/public_/authorize_jsp.class
/opt/zimbra/jetty_base/work/zimbra/jsp/org/apache/jsp/public_/authorize_jsp.java
```

**Reply by ianw1974** [reply]
The previous critical issue they had when files needed to be removed, rebuilding didn't help, since the commits hadn't been made to their repositories to allow for this.  This subsequently meant I had to script removing during the build process until Synacor/Zimbra finally commited the changes to their repositories which no longer meant such action didn't need to be taken.

In that instance, I manually took control of this at the end of August.  The commits weren't made by Zimbra until a month later to address that (end of September).  I will generally only intervene in such a way if the issue is critical.  Therefore waiting a week or so potentially means they will have ensured their repositories are up-to-date before a build is made.  Otherwise, a build is made that when later examined actually hasn't addressed the problem.  Also during that situation, the information was withheld unless you were a paying client.

Obviously high issues need to be addressed.  I also have to find time to make builds, check/test to see if they did actually resolve the problem or not.  Potentially then have to manually intervene and get my scripts to take care of it otherwise (if that is even possible).  If the fixes rely on updating components of Zimbra, such as Java or whatever, and the commits haven't been made, that is way too much work to undertake to even attempt to do that manually outside of Zimbra's commits.  I don't have time to for that level of involvement.

Waiting a week or so for Zimbra to make those commits at least makes my life easier by them ensuring that they've actually pushed everything to their repositories.

Had I just pushed a release days before they fixed it, I would most likely just generate new builds anyway.  Since it's pretty much days before my next quarterly build, I don't feel their is too much of a rush.

Making builds takes time.  As does the process of manually intervening to ensure it has been fixed outside of Zimbra's repositories until the commits are made.  Then updating the scripts to take care of that, building, testing, pushing the builds to the website.  I'm only one person doing this, so sometimes it's just a case of be patient :)

The scripts will build 10.0.6 anyway, so if anyone cannot wait until then, they have the choice to build themselves :)

**Comment by jeredfloyd** [comment]
Another remedial question as I'm not familiar with the build system:  I was surprised (but perhaps not really) that performing a "dnf update" on my zimbra host pulled the new Zimbra OpenJDK build, because I am still using the Zimbra repos for various support components.

Is this the wrong way to run third-party builds?  Will Zimbra continue to update and support FOSS components?  I have 223 packages installed with "zimbra" in the name, but only 27 of them are part of the install bundle. (Even more concerning is that 168 of those packages have "zimbra8.7" in the package version name...)

**Comment by ianw1974** [comment]
Zimbra 8 will pull in patches.  Zimbra 9 and 10 will not - at least as far as I am aware since the Zimbra repos now are only for the Zimbra NE release.  This means for anyone running 9 and 10 requires downloading a new build and running ```./install.sh```

**Reply by jeredfloyd** [reply]
This wasn't the case for me; they don't seem to have enabled any access control on the repos yet. 

When I run install.sh and it asks "Use Zimbra's package repository" I should say no? Where do all of the other packages pull from in that case?

**Reply by jeredfloyd** [reply]
To clarify, I'm not talking about the "-patch" packages, but the many open source components that Zimbra rebuilds and pins.  For example, I just got:
```
2023-12-20T17:14:40+0000 DEBUG Upgraded: zimbra-core-components-4.0.2-1zimbra10.0b1.el8.x86_64
2023-12-20T17:14:40+0000 DEBUG Upgraded: zimbra-ldap-components-3.0.2-1zimbra10.0b1.el8.x86_64
2023-12-20T17:14:40+0000 DEBUG Upgraded: zimbra-openjdk-17.0.8-1zimbra8.8b1.el8.x86_64
2023-12-20T17:14:40+0000 DEBUG Upgraded: zimbra-openjdk-cacerts-1.0.10-1zimbra8.7b1.el8.x86_64
```

(The 8.7 and 8.8 versioning seems to be an artifact, perhaps of packages that have not changed incompatibly since 8.7)

**Comment by ianw1974** [comment]
Saying yes/no to the repositories apparently configures the yum/dnf/apt repo files - check the Zimbra forums for that.  There are posts about that already.  Better would also be to ask there as well about the other packages that you get from those repositories.  As I mentioned, as far as I am aware, patches plus other updates come for Zimbra 8 - just like your example above.  They do not however get updated for Zimbra 9 or 10.

Ask on the forums as Synacor/Zimbra will have to answer that for you.  This place is for talking about my Zimbra build scripts, and/or about my builds appearing on the website :)

I don't offer Zimbra mail server support - that is what the official Zimbra forums is for :)

**Reply by jeredfloyd** [reply]
Sounds good; I'll bring up this topic on the "[Zimbra 10 FOSS Installation Guide](https://forums.zimbra.org/viewtopic.php?t=72231)" thread.
